aten/src/ATen/templates/TensorBody.h

```diff
@@ -599,6 +599,21 @@ class CAFFE2_API Tensor {
     return impl_->grad();
   }
 
+  // The Forward AD API functions below are low level and are not to be used by end
+  // users who should use the API provided in torch/csrc/autograd.h
+
+  /// This function returns the forward gradient for this Tensor at the given level.
+  const Tensor& fw_grad(uint64_t level) const {
+    return impl_->fw_grad(level, *this);
+  }
+
+  /// This function can be used to set the value of the forward grad.
+  /// Note that the given value might not be used directly if it is a view of another Tensor.
```

Comment:
What does this mean? Does that mean that it is silently discarded if `self` is a view?